nntrainer/tensor/manager.cpp

```diff
@@ -547,6 +556,8 @@ void Manager::initializeTensors(bool trainable) {
     }
     use_first_last = 1 - use_first_last;
   }
+
+  tensors_initialized = true;
```

Comment:
tensors_initialized includes gradients so from the invariant defined in the header. If `trainable` is false, `tensor_initialized` is nor false or true. I think it'd better to have a bit flag or equivalent to tell which one is initialized or not.

Reply:
The reason the init variables are not separated is that all 3 tensors (inouts, gradients, and derivatives) are all clubbed together in the abstraction exposed by the manager.
For internal purposes, I will add a variable to store the state of the trainable used for initialization, which will also match with the exposed abstraction provided by the manager.

Reply:
I will do this in a separate PR as this introduces some big changes, when considering about inference/train being called again and again.

Reply:
Sound decent.

Reply:
Added #981.